Rubberduck.Inspections/InspectionResult.cs

```diff
@@ -9,32 +9,37 @@
 using Rubberduck.UI.Controls;
 using Rubberduck.VBEditor;
 
-namespace Rubberduck.Inspections.Abstract
+namespace Rubberduck.Inspections
 {
-    public abstract class InspectionResultBase : IInspectionResult, INavigateSource, IExportable
+    public class InspectionResult : IInspectionResult, INavigateSource, IExportable
     {
-        protected InspectionResultBase(IInspection inspection, Declaration target)
-            : this(inspection, target.QualifiedName.QualifiedModuleName, target.Context, target)
+        public InspectionResult(IInspection inspection, string description, Declaration target)
+            : this(inspection, description, new QualifiedContext<ParserRuleContext>(target.QualifiedName.QualifiedModuleName, target.Context), target)
         { }
-        
-        protected InspectionResultBase(IInspection inspection, QualifiedModuleName qualifiedName, QualifiedMemberName? qualifiedMemberName, ParserRuleContext context)
+
+        public InspectionResult(IInspection inspection, string description, QualifiedContext context, QualifiedMemberName? qualifiedMemberName)
         {
             Inspection = inspection;
-            QualifiedName = qualifiedName;
+            Description = description.Capitalize();
+            QualifiedName = context.ModuleName;
             QualifiedMemberName = qualifiedMemberName;
-            Context = context;
+            Context = context.Context;
         }
-        
-        protected InspectionResultBase(IInspection inspection, QualifiedModuleName qualifiedName, ParserRuleContext context, Declaration target)
+
+        public InspectionResult(IInspection inspection, string description, QualifiedContext context, Declaration target, bool navigateToTarget = true)
         {
             Inspection = inspection;
-            QualifiedName = qualifiedName;
-            Context = context;
+            Description = description.Capitalize();
+            QualifiedName = context.ModuleName;
+            Context = context.Context;
             Target = target;
+            NavigateToTarget = navigateToTarget;
 
             QualifiedMemberName = GetQualifiedMemberName(target);
         }
 
+        private bool NavigateToTarget { get; } = false;
```

Comment:
It can still navigate--it just uses the `Context` instead of the `Target` if this is `false.